keyboards/yandrstudio/bl75/bl75.c

```diff
@@ -0,0 +1,139 @@
+    },
+    {
+      4, 4, 4, 4, 4, 4, 4, 4, 4, 4,
+      4, 4, 4, 4, 4, 4, 4, 4, 4, 4,
+      4, 4, 4, 4, 4, 4, 4, 4, 4, 4,
+      4, 4, 4, 4, 4, 4, 4, 4, 4, 4,
+      4, 4, 4, 4, 4, 4, 4, 4, 4, 4,
+      4, 4, 4, 4, 4, 4, 4, 4, 4, 4,
+      4, 4, 4, 4, 4, 4, 4, 4, 4, 4,
+      4, 4, 4, 4, 4, 4, 4, 4, 4, 4,
+      4, 4, 4, 4,
+      2, 2, 2, 2, 2, 2, 2, 2, 2, 2,
+      2, 2, 2, 2, 2, 2, 2, 2, 2, 2,
+      2, 2, 2, 2, 2, 2, 2, 2, 2, 2,
+      2, 2
+    }
+};
+
+
+
+bool rgb_matrix_indicators_advanced_kb(uint8_t led_min, uint8_t led_max) {
+    if (!rgb_matrix_indicators_advanced_user(led_min, led_max)) {
+        return false;
+    }
+    if (rgb_matrix_is_enabled()) {
+        if (kb_config.underground_rgb_sw == 1) {
+            for (uint8_t i = led_min; i < led_max; ++i) {
+                if ((g_led_config.flags[i] == 4)) {
+                    rgb_matrix_set_color(i, 0, 0, 0);
+                }
+            }
+        } else if (kb_config.underground_rgb_sw == 2) {
+            for (uint8_t i = led_min; i < led_max; ++i) {
+                if ((g_led_config.flags[i] == 2)) {
+                    rgb_matrix_set_color(i, 0, 0, 0);
+                }
+            }
+        }
+    } else {
+        rgb_matrix_set_color_all(0, 0, 0);
+    }
+    return true;
+}
+
+void eeconfig_init_kb(void) {
+    kb_config.raw = 0;
+    eeconfig_update_kb(kb_config.raw);
+    eeconfig_init_user();
+}
+
+void keyboard_post_init_kb(void) {
+    kb_config.underground_rgb_sw = eeconfig_read_kb();
+    rgb_matrix_reload_from_eeprom();
+    keyboard_post_init_user();
+}
+
+#endif
+
+
+bool process_record_kb(uint16_t keycode, keyrecord_t *record) {
+    if (!process_record_user(keycode, record)) {
+        return false;
+    }
+    switch (keycode) {
+#ifdef RGB_MATRIX_ENABLE
+        case RGB_KG_T:
+            if (rgb_matrix_config.enable && record->event.pressed) {
+                kb_config.underground_rgb_sw += 1;
+                kb_config.underground_rgb_sw %= 3;
+                eeconfig_update_kb(kb_config.raw);
+            }
+            return false;
+#endif
+        case KC_LG:
+            if (record->event.pressed) {
+                process_magic(GUI_TOG, record);
+            }
+            return false;
+        case KC_MACOS:
+            if (record->event.pressed) {
+                process_magic(CG_TOGG, record);
+            }
+            return false;
+        case KC_MCTL:
+            if (record->event.pressed) {
+                host_consumer_send(0x29F);
+            } else {
+                host_consumer_send(0);
+            }
+            return false;
+        case KC_LPAD:
+            if (record->event.pressed) {
+                host_consumer_send(0x2A0);
+            } else {
+                host_consumer_send(0);
+            }
+            return false;
+        default:
+            return true;
+    }
+    return true;
+}
```

Comment:
This allows for a default behavior to be defined, without the need for the user to specify the behavior.  This is very useful for the configurator which doesn't have a way to configure this behavior, currently.

```suggestion
}

#ifdef ENCODER_ENABLE
bool encoder_update_kb(uint8_t index, bool clockwise) {
    if (!encoder_update_user(index, clockwise)) { return false; }
    if (index == 0) {
        if (clockwise) {
            tap_code_delay(KC_VOLU, 10);
        } else {
            tap_code_delay(KC_VOLD, 10);
        }
    }
    return true;
}
#endif
```